app/src/main/java/com/example/tasky/authentication/domain/NameValidator.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.tasky.authentication.domain
+
+import javax.inject.Inject
+
+class NameValidator @Inject constructor() {
+
+    fun validateName(name: String) =
```

Comment:
Since NameValidator will always validate name, no need to be redundant in the function name

You could either:
1- Name it "validate"
2- Override the operator invoke function (operator fun invoke(name: String))


Same with Email and Password